tensorboard/plugins/pr_curve/pr_curve_demo.py

```diff
@@ -0,0 +1,198 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Create sample PR curve summary data.
+
+We have 3 classes: R, G, and B. We generate colors within RGB space from 3
+normal distributions (1 at each corner of the color triangle: [255, 0, 0],
+[0, 255, 0], and [0, 0, 255]).
+
+The true label of each random color is associated with the normal distribution
+that generated it.
+
+Using 3 other normal distributions (over the distance eac color is from a corner
+of the color triangle - RGB), we then compute the probability that each color
+belongs to the class. We use those probabilities to generate PR curves.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import math
+import os.path
+
+from six.moves import xrange  # pylint: disable=redefined-builtin
+import tensorflow as tf
+
+from tensorboard.plugins.pr_curve import summary
+
+FLAGS = tf.flags.FLAGS
+
+tf.flags.DEFINE_string('logdir', '/tmp/pr_curve_demo',
+                       'Directory into which to write TensorBoard data.')
+
+tf.flags.DEFINE_integer('steps', 10,
+                        'Number of steps to generate for each PR curve. '
+                        'Should be less than 40 for reasons in the code.')
```

Comment:
It would also be quite easy to lift this restriction by parameterizing `40` over `steps`.